src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -1087,6 +1108,127 @@ private static FunctionResult LoadJsonContentResultBuilder(IBinder binder, IFile
             return new(ConvertJsonToBicepType(token), ConvertJsonToExpression(token));
         }
 
+        private static FunctionResult LoadYamlContentResultBuilder(IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, FunctionCallSyntaxBase functionCall, ImmutableArray<TypeSymbol> argumentTypes)
+        {
+            var arguments = functionCall.Arguments.ToImmutableArray();
+            string? tokenSelectorPath = null;
+            if (arguments.Length > 1)
+            {
+                if (argumentTypes[1] is not StringLiteralType tokenSelectorType)
+                {
+                    return new(ErrorType.Create(DiagnosticBuilder.ForPosition(arguments[1]).CompileTimeConstantRequired()));
+                }
+                tokenSelectorPath = tokenSelectorType.RawStringValue;
+            }
+            if (!TryLoadTextContentFromFile(binder, fileResolver, diagnostics,
+                    (arguments[0], argumentTypes[0]),
+                    arguments.Length > 2 ? (arguments[2], argumentTypes[2]) : null,
+                    out var fileContent,
+                    out var errorDiagnostic,
+                    LanguageConstants.MaxJsonFileCharacterLimit))
+            {
+                return new(ErrorType.Create(errorDiagnostic));
+            }
+
+            if (ExtractTokenFromObject(fileContent) is not { } token)
+            {
+                // Instead of catching and returning the JSON parse exception, we simply return a generic error.
+                // This avoids having to deal with localization, and avoids possible confusion regarding line endings in the message.
+                return new(ErrorType.Create(DiagnosticBuilder.ForPosition(arguments[0]).UnparseableJsonType()));
+            }
+
+            if (tokenSelectorPath is not null)
+            {
+                try
+                {
+                    var selectTokens = token.SelectTokens(tokenSelectorPath, false).ToList();
+                    switch (selectTokens.Count)
+                    {
+                        case 0:
+                            return new(ErrorType.Create(DiagnosticBuilder.ForPosition(arguments[1]).NoJsonTokenOnPathOrPathInvalid()));
+                        case 1:
+                            token = selectTokens.First();
+                            break;
+                        default:
+                            token = new JArray();
+                            foreach (var selectToken in selectTokens)
+                            {
+                                ((JArray)token).Add(selectToken);
+                            }
```

Comment:
Casts are best avoided when possible as the lack of type-safety makes future refactoring error-prone.

```suggestion
                            var arrayToken = new JArray();
                            token = arrayToken;
                            foreach (var selectToken in selectTokens)
                            {
                                arrayToken.Add(selectToken);
                            }
```

Reply:
@anthony-c-martin ,

Chat-GPT recommended I refactor that section to simply

`token = new JArray(selectTokens);`

The unit tests will tell me for sure, but does that look right to you?